sync/src/chain.rs

```diff
@@ -162,6 +162,8 @@ struct PeerInfo {
 	asking: PeerAsking,
 	/// A set of block numbers being requested
 	asking_blocks: Vec<BlockNumber>,
+	/// Holds requested header hash if currently requesting block header by hash
+	asking_hash: Option<H256>,
```

Comment:
Normally headers and bodies are downloaded by number, unless downloading a header known from NewBlockHashes packet. 
Currently, if two NewBlockHash packets with the same hash arrive at the same time this may result in same header being requested twice.
This was added to keep track of the requested hash.